### PR TITLE
Create abs, absDot and Correct functions for Vector2d

### PR DESCRIPTION
**Pull request by ximenesfel**
- Abs() => Get the absolute value of the vector
- AbsDot() => The absolute dot product
- Correct() => Corrects any nan values

Create unit test for all functions.

Did all checks and tests.

These functions intend to solve issue #71

Signed-off-by: Felipe Ximenes <ximenesfel@gmail.com>

### CODE REVIEW

**Comment by ximenesfel** [comment]
After creating the pull request the following CI checks does not passed:

**Ticket opened / Add ticket to inbox (pull_request)** 

```
Action:   opened
sha:      d641c46767ef7b5c23c60ce4562043eed42fba1c
ref:      refs/pull/143/merge
Labels:
owner:    ignitionrobotics
repo:     ign-math

Dump context
Dump Payload
==================================================

Error: Input required and not supplied: GITHUB_TOKEN
    at Object.getInput (/home/runner/work/_actions/technote-space/create-project-card-action/v1/node_modules/@actions/core/lib/core.js:78:15)
    at Object.exports.getAccessToken (/home/runner/work/_actions/technote-space/create-project-card-action/v1/node_modules/@technote-space/github-action-helper/dist/utils.js:66:47)
    at Object.exports.getOctokit (/home/runner/work/_actions/technote-space/create-project-card-action/v1/node_modules/@technote-space/github-action-helper/dist/utils.js:67:106)
    at /home/runner/work/_actions/technote-space/create-project-card-action/v1/lib/main.js:27:66
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/technote-space/create-project-card-action/v1/lib/main.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/runner/work/_actions/technote-space/create-project-card-action/v1/lib/main.js:4:12)
    at run (/home/runner/work/_actions/technote-space/create-project-card-action/v1/lib/main.js:19:19)
    at Object.<anonymous> (/home/runner/work/_actions/technote-space/create-project-card-action/v1/lib/main.js:29:1)
##[error]Input required and not supplied: GITHUB_TOKEN
```

**Ubuntu Bionic CI / Ubuntu Bionic CI (pull_request)**

`Unexpected input(s) 'script-after-make', valid inputs are ['entryPoint', 'args', 'github_token', 'apt-dependencies', 'codecov-token', 'cmake-args']`

Looking the errors messages, I can't found the reason why the checks failed.

**Comment by chapulina** [comment]
> I can't found the reason why the checks failed.

That's a problem with our infrastructure, don't worry about it.

File: include/ignition/math/Vector2.hh [hunk 1]
```diff
@@ -134,6 +134,39 @@ namespace ignition
...
+                       std::abs(this->data[1]));
+      }
+
+      /// \brief Return the absolute dot product of this vector and
+      /// another vector. This is similar to the Dot function, except the
+      /// absolute value of each component of the vector is used.
+      ///
+      /// result = abs(x1 * x2) + abs(y1 * y2)
+      ///
+      /// \param[in] _v the vector
```
Comment (ahcorde) [inline include/ignition/math/Vector2.hh:151]: nit: Upper case after argument name

```suggestion
      /// \param[in] _v The vector
```

Comment (ximenesfel) [reply include/ignition/math/Vector2.hh:151]: @ahcorde Thanks for your comment.

**Pull request merged**